collect_app/src/main/java/org/odk/collect/android/fragments/dialogs/RankingWidgetDialog.java

```diff
@@ -35,57 +36,46 @@
 
 import org.javarosa.core.model.FormIndex;
 import org.javarosa.core.model.SelectChoice;
+import org.jetbrains.annotations.NotNull;
 import org.odk.collect.android.R;
 import org.odk.collect.android.R.string;
 import org.odk.collect.android.adapters.RankingListAdapter;
+import org.odk.collect.android.fragments.viewmodels.RankingViewModel;
 import org.odk.collect.android.utilities.QuestionFontSizeUtils;
 import org.odk.collect.android.utilities.RankingItemTouchHelperCallback;
 
-import java.io.Serializable;
+import java.util.ArrayList;
 import java.util.List;
 
 public class RankingWidgetDialog extends DialogFragment {
-
-    private static final String ITEMS = "items";
-    private static final String FORM_INDEX = "form_index";
-
     private RankingListener listener;
-
     private RankingListAdapter rankingListAdapter;
     private List<SelectChoice> items;
     private FormIndex formIndex;
+    private RankingViewModel viewModel;
 
     public interface RankingListener {
         void onRankingChanged(List<SelectChoice> items);
     }
 
-    public static RankingWidgetDialog newInstance(List<SelectChoice> items, FormIndex formIndex) {
-        RankingWidgetDialog dialog = new RankingWidgetDialog();
-        Bundle bundle = new Bundle();
-        bundle.putSerializable(ITEMS, (Serializable) items);
-        bundle.putSerializable(FORM_INDEX, formIndex);
-        dialog.setArguments(bundle);
+    public RankingWidgetDialog() {
+    }
 
-        return dialog;
+    public RankingWidgetDialog(List<SelectChoice> items, FormIndex formIndex) {
+        this.items = new ArrayList<>(items);
+        this.formIndex = formIndex;
```

Comment:
I don't love that these fields still end up existing since the point of the view model is to host them. What I was imagining originally is the activity creating the view model and the dialog fragment accessing it. I wasn't thinking about how the activity in this case is `FormEntryActivity` but currently it's the widget's responsibility to know about the items. 

Given the existing infrastructure, I find this ok for now. It's something @SaumiaSinghal and @seadowg could give another look at when they get to this widget.

Reply:
>What I was imagining originally is the activity creating the view model and the dialog fragment accessing it

we can't create viewmodels using activity because then if we have multiple rank widgets they will share the same viewmodel with the same data. We need to create viewmodels using fragment scope.

I didn't think about it much so maybe I'm wrong but generally it would be great if every single widget were a fragment then we could have a viewmodel for every widget easily.